### PR TITLE
inherit key "verify" from env into session so that kv_client can read it properly

### PR DESCRIPTION
**Pull request by chz8494**
---

Currently Hashicorp Vault as secret backend against private vault doesn't work and facing ssl CA trust issue.
Example config:
```
[secrets]
backend = airflow.providers.hashicorp.secrets.vault.VaultBackend
backend_kwargs = {"connections_path": "airflow-connections", "variables_path": null, "mount_point": "airflow", "url": "http://127.0.0.1:8200", "verify": "/etc/ssl/certs/ca-certificates"}
```
current code ignores `verify` value and causing CA trust failure.
I did some debug and found that the dep kv client tends to inherit session keys, so I just added `verify` value into session.

### CODE REVIEW

File: airflow/providers/hashicorp/_internal_client/vault_client.py [hunk 1]
```diff
@@ -207,6 +207,9 @@ def _client(self) -> hvac.Client:
             session = Session()
             session.mount("http://", adapter)
             session.mount("https://", adapter)
+            if self.kwargs is not None and "verify" in self.kwargs:
```
Comment (dirrao) [inline airflow/providers/hashicorp/_internal_client/vault_client.py]:
```suggestion
            if self.kwargs and "verify" in self.kwargs:
```

File: airflow/providers/hashicorp/_internal_client/vault_client.py [hunk 2]
```diff
@@ -207,6 +207,9 @@ def _client(self) -> hvac.Client:
...
+                if self.kwargs["verify"] is not None:
```
Comment (dirrao) [inline airflow/providers/hashicorp/_internal_client/vault_client.py]:
```suggestion
                if self.kwargs["verify"]:
```

**Review by dirrao — CHANGES_REQUESTED** [review]

**Review by hussein-awala — COMMENTED** [review]
The verify argument is present in [Airflow documentation ](https://github.com/apache/airflow/blob/8b723e498de47a14f4d2dc13d5a2b876d42d7ff8/docs/apache-airflow-providers-hashicorp/secrets-backends/hashicorp-vault.rst#vault-running-with-self-signed-certificates), but looks like it was accidentally removed by a change, so we need to add a test to avoid removing your fix in the future; could you add the test?
https://github.com/apache/airflow/blob/3568b09e8c7501bd84d08d594038dea9b8e20a23/tests/providers/hashicorp/_internal_client/test_vault_client.py

**Comment by chz8494** [comment]
> The verify argument is present in [Airflow documentation ](https://github.com/apache/airflow/blob/8b723e498de47a14f4d2dc13d5a2b876d42d7ff8/docs/apache-airflow-providers-hashicorp/secrets-backends/hashicorp-vault.rst#vault-running-with-self-signed-certificates), but looks like it was accidentally removed by a change, so we need to add a test to avoid removing your fix in the future; could you add the test? https://github.com/apache/airflow/blob/3568b09e8c7501bd84d08d594038dea9b8e20a23/tests/providers/hashicorp/_internal_client/test_vault_client.py

https://github.com/apache/airflow/blob/d3dc88f0844bcb377a9e52312e1a99b5ca6e617e/tests/providers/hashicorp/_internal_client/test_vault_client.py#L615
I'd like to help but don't understand how to run this test, seems it needs to be used against local vault server localhost:8081? but I don't know how it can be used on git-ci...
In addition, I traced back up to 2.1.0 but couldn't find any code related to `verify`, I'd assume this `session` change was new in recent kv_client update and `verify` logic in provider was never in place, someone coded it and expected **kwargs could automatically cover `verify` key value, which it's not happening.

see kv client doc and code:
https://github.com/hvac/hvac/blob/48027db42c037fe8f6b1c6fd8f6dbf80c1ea8595/docs/advanced_usage.rst#L108
https://github.com/hvac/hvac/blob/48027db42c037fe8f6b1c6fd8f6dbf80c1ea8595/hvac/adapters.py#L90

**Comment by hussein-awala** [comment]
> I'd like to help but don't understand how to run this test, seems it needs to be used against local vault server localhost:8081?

They are unit tests and not integration tests, so we use Python mock package to patch the methods/classes.
You can check [the testing docs](https://github.com/apache/airflow/blob/main/contributing-docs/09_testing.rst) in [the contributing documentation](https://github.com/apache/airflow/blob/main/contributing-docs).

> In addition, I traced back up to 2.1.0 but couldn't find any code related to verify, I'd assume this session change was new in recent kv_client update and verify logic in provider was never in place, someone coded it and expected **kwargs could automatically cover verify key value, which it's not happening.

This is possible, but we need to add tests to avoid breaking the fix in the future.

**Comment by chz8494** [comment]
@hussein-awala not sure if I'm doing it correctly, just added ssl case for v1 test. Also noticed a corner case problem with condition check, so I also change code to be:
```pythton
if self.kwargs and "verify" in self.kwargs:
                session.verify = self.kwargs["verify"]
```

**Comment by chz8494** [comment]
@hussein-awala can someone review and suggest if tests good?

**Review by hussein-awala — APPROVED** [review]

**Comment by eladkal** [comment]
Static tests are failing. Can you look into it?

**Comment by chz8494** [comment]
> Static tests are failing. Can you look into it?

fixed

**Comment by chz8494** [comment]
@hussein-awala @eladkal can someone help merge it? I'd like to see it works in the new release

**Review by hussein-awala — APPROVED** [review]

**Comment by hussein-awala** [comment]
Congrats on your first commit 🎉

**Comment by melicheradam** [comment]
@hussein-awala Hello, any updates on when this will get released? Since it was merged on 15th may but was not included in the `2.9.2` version

**Comment by potiuk** [comment]
> @hussein-awala Hello, any updates on when this will get released? Since it was merged on 15th may but was not included in the 2.9.2 version

It's provider change - look at provider's changelogs and releases. What do you make of it @melicheradam ?

**Comment by melicheradam** [comment]
@potiuk thanks, I was not aware that providers have their separate changelogs and releases :)

**Pull request merged**